### PR TITLE
chore: auto doc index endpoints

### PR DESCRIPTION
**Pull request by ryanfang5**

### CODE REVIEW

**Comment by ryanfang5** [comment]
Moved the post method to a create an index into references/api. This method was under the endpoint "/refs/{ref_id}/indexes" which is already defined as an endpoint in references.

File: virtool/indexes/api.py [hunk 1]
```diff
@@ -31,89 +32,100 @@
 routes = Routes()
 
 
-@routes.get("/indexes")
-async def find(req):
-    """
-    Return a list of indexes.
+@routes.view("/indexes")
+class IndexesView(PydanticView):
+    async def get(self) -> Union[r200[GetIndexesResponse]]:
```
Comment (igboyes) [inline virtool/indexes/api.py:37]:
```suggestion
    async def get(self) -> Union[r200[ListIndexesResponse]]:
```

File: virtool/indexes/api.py [hunk 2]
```diff
@@ -31,89 +32,100 @@
...
+        """
+        Find indexes.
 
-    """
-    db = req.app["db"]
+        Return a list of indexes.
```
Comment (igboyes) [inline virtool/indexes/api.py:41]:
```suggestion
        Retrieves a list of indexes.
```

File: virtool/indexes/api.py [hunk 3]
```diff
@@ -31,89 +32,100 @@
...
-        {"$sort": {"version": -1}},
-        {
-            "$group": {
-                "_id": "$reference.id",
-                "index": {"$first": "$_id"},
-                "version": {"$first": "$version"},
-            }
-        },
-    ]
+        ready = self.request.query.get("ready", False)
```
Comment (igboyes) [inline virtool/indexes/api.py:49]: Use `aiohttp-pydantic` support for query strings.

https://github.com/Maillol/aiohttp-pydantic#inject-query-string-parameters

File: virtool/indexes/api.py [hunk 4]
```diff
@@ -31,89 +32,100 @@
...
+            200: Successful operation
+            404: Not found
 
-    index_id = req.match_info["index_id"]
+        """
+        db = self.request.app["db"]
+        pg = self.request.app["pg"]
 
-    document = await db.indexes.find_one(index_id)
+        index_id = self.request.match_info["index_id"]
```
Comment (igboyes) [inline virtool/indexes/api.py:102]: Use `aiohttp-pydantic` support for path parameter validation.

https://github.com/Maillol/aiohttp-pydantic#inject-path-parameters

File: virtool/indexes/api.py [hunk 5]
```diff
@@ -154,42 +166,47 @@ async def download_otus_json(req):
...
+        Download files relating to a given index.
 
-    if filename not in FILES:
-        raise NotFound()
+        Status Codes:
+            200: Successful operation
+            404: Not found
+        """
+        index_id = self.request.match_info["index_id"]
+        filename = self.request.match_info["filename"]
```
Comment (igboyes) [inline virtool/indexes/api.py:180]: https://github.com/Maillol/aiohttp-pydantic#inject-path-parameters

File: virtool/indexes/api.py [hunk 6]
```diff
@@ -379,36 +329,43 @@ async def finalize(req):
     return json_response(await virtool.indexes.db.processor(db, document))
 
 
-@routes.get("/indexes/{index_id}/history")
-async def find_history(req):
-    """
-    Find history changes for a specific index.
+@routes.view("/indexes/{index_id}/history")
+class IndexHistoryView(PydanticView):
+    async def get(self) -> Union[r200[GetHistoryResponse], r404]:
```
Comment (igboyes) [inline virtool/indexes/api.py:334]:
```suggestion
    async def get(self) -> Union[r200[ListHistoryResponse], r404]:
```

**Review by igboyes — CHANGES_REQUESTED** [review]

**Comment by ryanfang5** [comment]
Aiohttp-pydantic query validators will be done in the data layer.

**Review by igboyes — APPROVED** [review]

**Pull request merged**